data/libs/PublicFunctions.lua

```diff
@@ -0,0 +1,20 @@
+-- Public Functions
+
+-- spawn pirates
+function SpawnPirate(name,risk) 
+	local shipdef = shipdefs[Engine.rand:Integer(1,#shipdefs)]
+	local default_drive = Equipment.hyperspace['hyperdrive_'..tostring(shipdef.hyperdriveClass)]
+
+	local max_laser_size = shipdef.capacity - default_drive.capabilities.mass
+	local laserdefs = utils.build_array(utils.filter(
+		function (k,l) return l:IsValidSlot('laser_front') and l.capabilities.mass <= max_laser_size and l.l10n_key:find("PULSECANNON") end,
+			pairs(Equipment.laser)
+		))
+	local laserdef = laserdefs[Engine.rand:Integer(1,#laserdefs)]
+
+	name = Space.SpawnShipNear(shipdef.id, Game.player, 50, 100)
+	name:SetLabel(Ship.MakeRandomLabel())
+	name:AddEquip(default_drive)
+	name:AddEquip(laserdef)
+
```

Comment:
This makes use of a number of functions that needs to be imported. Basically anything to the right of the `=` not previously defined, like Engine, Space, and utils.

Also, `shipdefs` was defined somewhere from where you copied the code.

And as I mentioned, the function needs to return the ship it has spawned (what you call `name` here). It aught to be `local` and maybe called `ship`?

There's the [API documentation](http://eatenbyagrue.org/f/pioneer/codedoc/files/LuaBody-cpp.html) for support.